agg-ast/schema_derivation/src/match_schema_derivation.rs

```diff
@@ -390,6 +390,59 @@ fn match_derive_schema_for_op(
 
 impl MatchConstrainSchema for Expression {
     fn match_derive_schema(&self, state: &mut ResultSetState) -> Result<()> {
+        /// date operators such as $dateAdd return null if _any_ of their arguments are null.
```

Comment:
can ignore the macros themselves. One benefit to looking at a macro level at every function is identifying some redundancies 😓